src/app/shared/data-classes-list/data-classes-list.component.ts

```diff
@@ -168,7 +208,7 @@ export class DataClassesListComponent implements AfterViewInit {
             });
          }
       });
-      const promise = new Promise((resolve, reject) => {
+      const promise = new Promise<void>((resolve, reject) => {
```

Comment:
I think this wrapping of an observable into a promise should be refactored out. The actions done in the `promise.then()` should just be part of the `subscribe()` call.